src/Game/AccountManagement.java

```diff
@@ -114,6 +123,222 @@ public static void register(String user, String pass) {
+	        
+			String oldPass = Client.formatText(Panel.getControlText(Client.panelRecovery, Client.recoverOldPassInput), 20);
+			String newPass = Client.formatText(Panel.getControlText(Client.panelRecovery, Client.recoverNewPassInput), 20);
+			
+			StreamUtil.newPacket(8);
+	        Object buffer = StreamUtil.getStreamBuffer();
+	        StreamUtil.putLongTo(buffer, Util.username2hash(Panel.getControlText(Client.panelLogin, Client.loginUserInput)));
+			
+			// In 235 putRandom of "random.dat" is 24 bytes, but for 127 is expected 4 bytes.
+	        Object randBlock = StreamUtil.getNewBuffer(24);
+	        StreamUtil.putRandom(randBlock);
+	        byte[] randArr = StreamUtil.getBufferByteArray(randBlock);
+	        CRC16 sum = new CRC16();
+	        sum.update(randArr);
+	        int rand = (int) sum.getValue();
+	        StreamUtil.putIntTo(buffer, rand);
+			
+	        enc_cred_put(buffer, oldPass + newPass, Client.session_id);
+	        
+	        for (int i=0; i < 5; ++i) {
+	        	/** In RSC127 it would have sent an encrypted "hash" like of answers, but seems in later revisions
+	          	 * incl. up to RSC175 was changed to just send the encrypted answers, probably due to collisions
+	          	 * in RSC127 hasher function */
+	        	String answer = Panel.getControlText(Client.panelRecovery, Client.controlRecoveryInput[i]);
+	        	answer = Util.formatString(answer, 50);
+	        	StreamUtil.putByteTo(buffer, (byte)answer.length());
+	            enc_cred_put(buffer, answer, Client.session_id);
+	        }
+
+			StreamUtil.flushPacket();
+			
+			StreamUtil.readStream(); // Unknown what data this contained, client doesn't use it.
+			int response = StreamUtil.readStream();
+			Logger.Game("Recover response: " + response);
+			if (response == 0) {
+				Client.login_screen = Client.SCREEN_USERNAME_PASSWORD_LOGIN;
+				Client.setLoginMessage("", "Sorry, recovery failed. You may try again in 1 hour");
+				Client.failedRecovery = true;
+				return;
+			}
+
+			if (response == 1) {
+				Client.login_screen = Client.SCREEN_USERNAME_PASSWORD_LOGIN;
+				Client.setLoginMessage("", "Your pass has been reset. You may now use the new pass to login");
+				return;
+			}
+
+			Client.login_screen = Client.SCREEN_USERNAME_PASSWORD_LOGIN;
+			Client.setLoginMessage("", "Recovery failed! Attempts exceeded?");
+			return;
+		} catch (Exception e) {
+			e.printStackTrace();
+			Client.setLoginMessage("Check your internet settings", "Sorry! Unable to connect to server.");
+		}
+  }
+  
+  public static void sendPassChange(String oldPwd, String newPwd) {
+	  String oldPass = Client.formatText(oldPwd, 20); 
+	  String newPass = Client.formatText(newPwd, 20);
+	  
+	  StreamUtil.newPacket(25);
+	  
+	  Object buffer = StreamUtil.getStreamBuffer();
+	  enc_cred_put(buffer, oldPass + newPass, Client.session_id);
+	  
+	  StreamUtil.sendPacket();
+  }
+  
+  public static void sendRecoveryQuestions() {

```

Comment:
Should note above this that this is function is intended to be RSC175 compatible

Reply:
added